packages/esm-patient-orders-app/src/order-basket/general-order-type/general-order-type.component.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { type ComponentProps, useCallback, useEffect, useMemo, useState } from 'react';
+import { Button, Tile } from '@carbon/react';
+import classNames from 'classnames';
+import styles from './general-order-panel.scss';
+import {
+  AddIcon,
+  ChevronDownIcon,
+  ChevronUpIcon,
+  type DefaultWorkspaceProps,
+  useLayoutType,
+} from '@openmrs/esm-framework';
+import { useTranslation } from 'react-i18next';
+import {
+  type DrugOrderBasketItem,
+  launchPatientWorkspace,
+  useOrderBasket,
+  useOrderType,
+} from '@openmrs/esm-patient-common-lib';
+import OrderBasketItemTile from './order-basket-item-tile.component';
+import { prepOrderPostData } from './resources';
+import { type OrderTypeDefinition } from '../../config-schema';
+
+interface GeneralOrderTypeProps extends OrderTypeDefinition {
+  closeWorkspace: DefaultWorkspaceProps['closeWorkspace'];
+}
+
+const GeneralOrderType: React.FC<GeneralOrderTypeProps> = ({ orderTypeUuid, closeWorkspace, label }) => {
+  const { t } = useTranslation();
+  const isTablet = useLayoutType() === 'tablet';
+  const { orderType, isLoadingOrderType } = useOrderType(orderTypeUuid);
+
+  const { orders, setOrders } = useOrderBasket<DrugOrderBasketItem>(orderTypeUuid, prepOrderPostData);
+  const [isExpanded, setIsExpanded] = useState(orders.length > 0);
+  const {
+    incompleteOrderBasketItems,
+    newOrderBasketItems,
+    renewedOrderBasketItems,
+    revisedOrderBasketItems,
+    discontinuedOrderBasketItems,
+  } = useMemo(() => {
+    const incompleteOrderBasketItems: Array<DrugOrderBasketItem> = [];
+    const newOrderBasketItems: Array<DrugOrderBasketItem> = [];
+    const renewedOrderBasketItems: Array<DrugOrderBasketItem> = [];
+    const revisedOrderBasketItems: Array<DrugOrderBasketItem> = [];
+    const discontinuedOrderBasketItems: Array<DrugOrderBasketItem> = [];
+
+    orders.forEach((order) => {
+      if (order?.isOrderIncomplete) {
+        incompleteOrderBasketItems.push(order);
+      } else if (order.action === 'NEW') {
+        newOrderBasketItems.push(order);
+      } else if (order.action === 'RENEW') {
+        renewedOrderBasketItems.push(order);
+      } else if (order.action === 'REVISE') {
+        revisedOrderBasketItems.push(order);
+      } else if (order.action === 'DISCONTINUE') {
+        discontinuedOrderBasketItems.push(order);
+      }
+    });
+
+    return {
+      incompleteOrderBasketItems,
+      newOrderBasketItems,
+      renewedOrderBasketItems,
+      revisedOrderBasketItems,
+      discontinuedOrderBasketItems,
+    };
+  }, [orders]);
+
+  const openConceptSearch = () => {
+    closeWorkspace({
+      ignoreChanges: true,
+      onWorkspaceClose: () =>
+        launchPatientWorkspace('orderable-concept-workspace', {
+          orderTypeUuid,
+        }),
+    });
+  };
+
+  const openOrderForm = (order: DrugOrderBasketItem) => {
```

Comment:
Type here seems wrong...

Reply:
Also, these callbacks almost certainly would benefit from using `useCallback()`, especially as they are passed as props to multiple components (potentially).